### PR TITLE
Add Golioth API Training

### PR DESCRIPTION
**Pull request by szczys**
This PR adds a new training module focused on the Golioth REST API.

Note that the structure of the Developer Training site has now been drastically changes:
* Existing training has been moved into the magtag-training directory and branded as "Zephyr Training" 
  * Future work will update this training to use other boards
* API Training and Zephyr Training added to the top nav bar
  * The left nav bar will update to only show the entries for the selected training

### CODE REVIEW

File: website/docusaurus.config.js
Comment (ChrisGammell) [inline website/docusaurus.config.js:1]: Don't think I don't see you making the default color scheme dark! :-D

**Review by ChrisGammell — APPROVED** [review]
Changes look great! One small fix I will submit before merging (formatting on one of the API pages)

**Comment by szczys** [comment]
> Changes look great! One small fix I will submit before merging (formatting on one of the API pages)

Thanks for catching that. I actually meant to apply that style change to all of the pages and forgot. I'll include it in a commit before merging.

File: website/docusaurus.config.js
Comment (szczys) [reply website/docusaurus.config.js:1]: Oh yeah, just wanted to be sure the page looks right :smiley:

**Pull request merged**